cmd/drone-buildx-gcr/main.go

```diff
@@ -1,63 +1,159 @@
+	var config Config
+
+	// Load env-file if it exists
 	if env := os.Getenv("PLUGIN_ENV_FILE"); env != "" {
-		godotenv.Load(env)
+		if err := godotenv.Load(env); err != nil {
+			log.Fatalf("Error loading .env file: %v", err)
+		}
 	}
 
-	var (
-		repo     = getenv("PLUGIN_REPO")
-		registry = getenv("PLUGIN_REGISTRY")
-		password = getenv(
+	idToken := getenv("PLUGIN_OIDC_TOKEN_ID")
+	projectId := getenv("PLUGIN_PROJECT_NUMBER")
+	poolId := getenv("PLUGIN_POOL_ID")
+	providerId := getenv("PLUGIN_PROVIDER_ID")
+	serviceAccountEmail := getenv("PLUGIN_SERVICE_ACCOUNT_EMAIL")
+
+	if idToken != "" && projectId != "" && poolId != "" && providerId != "" && serviceAccountEmail != "" {
+		federalToken, err := gcp.GetFederalToken(idToken, projectId, poolId, providerId)
+		if err != nil {
+			logrus.Fatalf("PLUGIN_OIDC_TOKEN_ID: %s, PLUGIN_PROJECT_NUMBER: %s, PLUGIN_POOL_ID: %s, PLUGIN_PROVIDER_ID: %s, PLUGIN_SERVICE_ACCOUNT_EMAIL: %s", idToken, projectId, poolId, providerId, serviceAccountEmail)
+			logrus.Fatalf("Error (getFederalToken): %s", err)
+		}
+		accessToken, err := gcp.GetGoogleCloudAccessToken(federalToken, serviceAccountEmail)
+		if err != nil {
+			logrus.Fatalf("Error (getGoogleCloudAccessToken): %s", err)
+		}
+		config.AccessToken = accessToken
+	} else {
+		password := getenv(
 			"PLUGIN_JSON_KEY",
 			"GCR_JSON_KEY",
 			"GOOGLE_CREDENTIALS",
 			"TOKEN",
 		)
-	)
+		config.WorkloadIdentity = parseBoolOrDefault(false, getenv("PLUGIN_WORKLOAD_IDENTITY"))
+		config.Username, config.Password = setUsernameAndPassword(username, password, config.WorkloadIdentity)
+	}
+	config.Repo = getenv("PLUGIN_REPO")
+	config.Registry = getenv("PLUGIN_REGISTRY")
 
-	// decode the token if base64 encoded
-	decoded, err := base64.StdEncoding.DecodeString(password)
-	if err == nil {
-		password = string(decoded)
+	return config
+}
+
+func main() {
+	config := loadConfig()
+	if config.AccessToken != "" {
+		os.Setenv("ACCESS_TOKEN", config.AccessToken)
+	} else if config.Username != "" && config.Password != "" {
+		os.Setenv("DOCKER_USERNAME", config.Username)
+		os.Setenv("DOCKER_PASSWORD", config.Password)
 	}
 
 	// default registry value
-	if registry == "" {
-		registry = "gcr.io"
+	if config.Registry == "" {
+		config.Registry = "gcr.io"
 	}
 
 	// must use the fully qualified repo name. If the
 	// repo name does not have the registry prefix we
 	// should prepend.
-	if !strings.HasPrefix(repo, registry) {
-		repo = path.Join(registry, repo)
+	if !strings.HasPrefix(config.Repo, config.Registry) {
+		config.Repo = path.Join(config.Registry, config.Repo)
 	}
 
-	os.Setenv("PLUGIN_REPO", repo)
-	os.Setenv("PLUGIN_REGISTRY", registry)
-	os.Setenv("DOCKER_USERNAME", username)
-	os.Setenv("DOCKER_PASSWORD", password)
+	os.Setenv("PLUGIN_REPO", config.Repo)
+	os.Setenv("PLUGIN_REGISTRY", config.Registry)
+	os.Setenv("DOCKER_USERNAME", config.Username)
+	if config.AccessToken != "" {
+		os.Setenv("PLUGIN_ACCESS_TOKEN", config.AccessToken)
```

Comment:
Who is using this env variable? `drone-buildx` is expecting `ACCESS_TOKEN` which you already set on L85.

Reply:
removed the unused/extra params